xla/tests/dot_operation_test.cc

```diff
@@ -41,6 +41,118 @@ limitations under the License.
 namespace xla {
 namespace {
 
+// won't be needed in C++20
+inline bool starts_with(const std::string& a, const std::string& b) {
+  return a.substr(0,b.size()) == b;
+}
+
+class GradFlagVerifier
+{
+  size_t size_x_=0, size_y_=0, flags_=0;
+  bool trivial_ = false;
+public:
+  GradFlagVerifier(XlaBuilder& builder, XlaOp& x, XlaOp& y, PrecisionConfig& cfg) {
+    cfg = xla::PrecisionConfigDEFAULT();
+#if TENSORFLOW_USE_ROCM
+    flags_ = rand() & 3;
```

Comment:
Do I understand correctly that you pick a random flag value?

Reply:
Yes, would it be better to loop over all values (increasing execution time of the unit test 4x)?